src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -210,13 +210,13 @@ void zero_past_key_values(ov::InferRequest& request) {
     }
 }
 
-void prepare_decoder_with_past(ov::InferRequest& decoder_with_past, ov::InferRequest& decoder) {
+void prepare_decoder_with_past(ov::InferRequest& decoder_with_past, ov::InferRequest& decoder, const size_t init_ids_size) {
     // NB: Prepare attetion mask to be in a format [0, 0, 0, 1, 1, 1, 1, ..., 0, 1]
     // Mask should be inverted for decoder_with_past 
     auto attention_mask = decoder_with_past.get_tensor("attention_mask");
     auto* attention_mask_ptr = attention_mask.data<ov::float16>();
-    std::fill(attention_mask_ptr, attention_mask_ptr + 3u, 0);
-    std::fill(attention_mask_ptr + 3u, attention_mask_ptr + attention_mask.get_size() - 2, 1);
+    std::fill(attention_mask_ptr, attention_mask_ptr + init_ids_size, 0);
+    std::fill(attention_mask_ptr + init_ids_size, attention_mask_ptr + attention_mask.get_size() - 2, 1);
```

Comment:
Why `-2`?

Reply:
Btw, since we reshape model anyway, we don't need `attention_mask` at all, probably we may not apply transformation that expose this

Reply:
This function is for decoder_with_past model.
It should have attention mask should be in format [0, 0, 0, 1, 1, 1, ... 1, 0, 1]

You mean we don't need attention mask for decoder model or for both model? How will it work then for decoder with past model?

Reply:
There's already such comment here: 
https://github.com/openvinotoolkit/openvino.genai/blob/5ab58ca70dd2774595ad82768074c7a497aa9377/src/cpp/src/whisper_pipeline_static.cpp#L214-L215